### ISSUE TITLE
[NBug] Resetting unstaged part of conflicted file throws error

### ISSUE THREAD

**Issue by GrahamTheCoder** [gitextensions/gitextensions#10423]
### Environment

- Git Extensions 4.0.0.15569
- Build cba315df5366fca01240d93b6e5a5077fd5e6630
- Git 2.38.1.windows.1
- Microsoft Windows NT 10.0.22000.0
- .NET 6.0.11
- DPI 120dpi (125% scaling)

### Issue description

Exit code: 1
Command: C:\Program Files\Git\bin\git.exe
Arguments: checkout-index --index --force -- "src/BrandVue.FrontEnd/cypress/e2e/menuLinks.cy.ts"
Working directory: C:\Repos\Vue\

```
GitExtUtils.ExternalOperationException: git checkout-index: src/BrandVue.FrontEnd/cypress/e2e/menuLinks.cy.ts is unmerged
 ---> System.Exception: git checkout-index: src/BrandVue.FrontEnd/cypress/e2e/menuLinks.cy.ts is unmerged
   --- End of inner exception stack trace ---
   at GitCommands.ExecutableExtensions.GetOutputAsync(IExecutable executable, ArgumentString arguments, Byte[] input, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes)
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at GitCommands.ExecutableExtensions.GetBatchOutput(IExecutable executable, ICollection`1 batchArguments, Byte[] input, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes)
   at GitCommands.GitModule.ResetFiles(IReadOnlyList`1 files)
   at GitUI.CommandsDialogs.FormCommit.ResetSoftClick(Object sender, EventArgs e)
   at GitUI.CommandsDialogs.FormCommit.ResetSelectedFiles()
   at GitUI.CommandsDialogs.FormCommit.ExecuteCommand(Int32 cmd)
   at ResourceManager.GitExtensionsFormBase.ProcessHotkey(Keys keyData)
   at GitUI.CommandsDialogs.FormCommit.ProcessHotkey(Keys keyData)
   at ResourceManager.GitExtensionsFormBase.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.Control.PreProcessMessage(Message& msg)
   at System.Windows.Forms.Control.PreProcessControlMessageInternal(Control target, Message& msg)
   at System.Windows.Forms.Application.ThreadContext.PreTranslateMessage(MSG& msg)
```

When there's a merge conflict, clicking reset on the unstaged entry causes this error.
Expected: Reset the file anyway, I don't care what state git thinks its in, I just want to nuke it

### Steps to reproduce

1. Pull rebase with a change to a file that's changed in the pulled commits too
2. Open the commit window
3. Try to reset the unstaged part of the file
4. Boom.

Looks like the wrong git command was issued for this particular scenario (treating it as unstaged). It works when you reset the staged part, so that's the command you need.

### Did this work in previous version of GitExtensions?

Don't think so

**Comment by gerhardol** [comment]
Special handling is needed for .IsConflict in `ResetSoftClick()`.
git-reset followed by git-checkout seem to be required,

**Comment by RussKie** [comment]
@mstv was #10556 targeting this as well?

@GrahamTheCoder can you please try one of the latest 4.0 build and see if your issue is fixed? E.g., this one https://ci.appveyor.com/project/gitextensions/gitextensions/builds/45820457/artifacts.

**Comment by GrahamTheCoder** [comment]
I can confirm it still repros on latest: 
[ReproGitExtensionsBug10423.zip](https://github.com/gitextensions/gitextensions/files/10356729/ReproGitExtensionsBug10423.zip)

![image](https://user-images.githubusercontent.com/2490482/210913153-90c56f2b-8a8c-465e-b70f-66b188c810e6.png)

- Git Extensions 4.0.2.16009
- Build 6fa3adcf4020ff8645747cbc7011a49eec9c54b0
- Git 2.39.0.windows.1
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.12
- DPI 120dpi (125% scaling)

Exit code: 1
Command: C:\Program Files\Git\bin\git.exe
Arguments: checkout-index --index --force -- "a.txt"
Working directory: C:\ReproGitExtensionsBug10423\

```
GitExtUtils.ExternalOperationException: git checkout-index: a.txt is unmerged
 ---> System.Exception: git checkout-index: a.txt is unmerged
   --- End of inner exception stack trace ---
   at GitCommands.ExecutableExtensions.GetOutputAsync(IExecutable executable, ArgumentString arguments, Byte[] input, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes)
   at Microsoft.VisualStudio.Threading.JoinableTask.CompleteOnCurrentThread()
   at GitCommands.ExecutableExtensions.GetBatchOutput(IExecutable executable, ICollection`1 batchArguments, Byte[] input, Encoding outputEncoding, CommandCache cache, Boolean stripAnsiEscapeCodes)
   at GitCommands.GitModule.ResetFiles(IReadOnlyList`1 files)
   at GitUI.CommandsDialogs.FormCommit.ResetSoftClick(Object sender, EventArgs e)
   at GitUI.CommandsDialogs.FormCommit.ResetSelectedFiles()
   at GitUI.CommandsDialogs.FormCommit.ExecuteCommand(Int32 cmd)
   at ResourceManager.GitExtensionsFormBase.ProcessHotkey(Keys keyData)
   at GitUI.CommandsDialogs.FormCommit.ProcessHotkey(Keys keyData)
   at ResourceManager.GitExtensionsFormBase.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.ContainerControl.ProcessCmdKey(Message& msg, Keys keyData)
   at ResourceManager.GitExtensionsControl.ProcessCmdKey(Message& msg, Keys keyData)
   at System.Windows.Forms.Control.PreProcessMessage(Message& msg)
   at System.Windows.Forms.Control.PreProcessControlMessageInternal(Control target, Message& msg)
   at System.Windows.Forms.Application.ThreadContext.PreTranslateMessage(MSG& msg)
```

**Comment by gerhardol** [comment]
@GrahamTheCoder Please try the changes in #10854
This handles the most common case with both modified, not files deleted and added

**Comment by GrahamTheCoder** [comment]
Looks like the CI build failed. I tried building from source, but don't have the right SDKs and couldn't build on the latest. Sorry I don't have time to do more - if the [ReproGitExtensionsBug10423.zip](https://github.com/gitextensions/gitextensions/files/10356729/ReproGitExtensionsBug10423.zip) now works, then consider this fixed. Thanks!

**Comment by gerhardol** [comment]
Restarted the build, that was a standard intermittent test case failure that we have spent many hours trying to fix.
The repro is fixed, both in FormCommit and Diff tab.

**Comment by GrahamTheCoder** [comment]
Nice one, thanks!

**Comment by gerhardol** [comment]
This is not a regression, the Git error was suppressed before.
Included in #10854 (with removal of the 'duplicate' conflict in #10852 soon).
Conflicts where files are added or removed will have to be handled manually (stage/unstage before reset).
